gum/backend-darwin/gumobjcapiresolver.c

```diff
@@ -176,8 +176,16 @@ gum_objc_api_resolver_enumerate_matches (GumApiResolver * resolver,
   gboolean carry_on;
   GHashTable * visited_classes;
   GumObjcClassMetadata * klass;
-
-  g_regex_match (self->query_pattern, query, 0, &query_info);
+  
+  gchar * query_copy = (gchar *) query;
+  int query_len = strlen(query_copy);
+  gboolean ignore_case = (query[query_len-1] == 'i' 
+                      && query[query_len-2] == '/');
+  if (ignore_case) {
+    query_copy = g_ascii_strdown (query, strlen (query));
```

Comment:
The `query_copy` string leaks when this branch is taken. Would suggest setting `query_copy = NULL` where it's declared, and assigning to `query` after this line, but leaving `query_copy` pointing at the copy. Then at the end of the scope do `g_free (query_copy)`, which is `NULL`-safe.